### PR TITLE
separate js from html, add dev server

### PR DESCRIPTION
**Pull request by WardCunningham**
First increment towards architecture of collaboration.

Worked with
- Daphne Watson
- Edwin Knuth

### CODE REVIEW

**Comment by WardCunningham** [comment]
This PR breaks the gh-pages but adds a good dev cycle of build, lint and reload.

Required additional work:
- [x] publish to gh-pages build step
- [x] document build in ReadMe

File: package.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "small-client-wiki",
+  "version": "1.0.0",
+  "description": "Read-Only React Federated Wiki Browser",
+  "main": "index.js",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/WardCunningham/small-client-wiki.git"
```
Comment (jasonrclark) [inline package.json:8]: Typo on the `git+`?

**Comment by jasonrclark** [comment]
Ran this locally, looking very nice so far. Love it popping the browser open when I `npm start`... hadn't seen that touch before.

File: package.json (hunk 1 above)
Comment (WardCunningham) [reply package.json:8]: npm interprets that strange arithmetic.
https://docs.npmjs.com/files/package.json

screen shot 2016-11-14 at 8 07 38 pm

Comment (jasonrclark) [reply package.json:8]: Ah, gotcha. I looked on that very page, but only in the section about the `"repository"` key. Learn something new every day!

**Comment by clemsos** [comment]
Now that there is a build system, it may be worth breaking jsx files into distinct components to make code reviewing and discovery easier.

File: package.json (hunk 1 above)
Comment (paul90) [reply package.json:8]: As this is a github repository, you could just say

```
"repository": "WardCunningham/small-client-wiki"
```

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+</style>
+
+<div id="content"></div>
+<script src="https://fb.me/react-0.14.6.js"></script>
+<script src="https://fb.me/react-dom-0.14.6.js"></script>
+<script src="http://fb.me/JSXTransformer-0.12.1.js"></script>
+<script src="https://code.jquery.com/jquery-3.1.0.js"></script>
+<script src="https://cdn.rawgit.com/showdownjs/showdown/1.4.3/dist/showdown.min.js"></script>
+<script type="text/jsx" src="index.js">
```
Comment (paul90) [inline public/index.html]: There is no need for

``` javascript
<script src="https://fb.me/react-0.14.6.js"></script>
<script src="https://fb.me/react-dom-0.14.6.js"></script>
<script src="http://fb.me/JSXTransformer-0.12.1.js"></script>

<script src="https://cdn.rawgit.com/showdownjs/showdown/1.4./dist/showdown.min.js"></script>
<script type="text/jsx" src="index.js">
```

as this is all dealt with by the build process.

Just need to keep the jquery load, for now, though I expect that will be replaced by using FETCH in due course.

**Comment by WardCunningham** [comment]
> If you’re getting started with React, use create-react-app to automate the build of your app. There is no configuration file, and react-scripts is the only extra build dependency in your package.json. Your environment will have everything you need to build a modern React app.

https://github.com/facebookincubator/create-react-app

We modeled the work in this pull request on the apps created with create-react-app. It promise no lock-in but will stop helping once you "eject".

File: public/index.html [hunk 3]
```diff
@@ -101,6 +101,5 @@
 
 <div id="content"></div>
 <script src="https://code.jquery.com/jquery-3.1.0.js"></script>
-<script type="text/jsx" src="index.js"></script>
```
Comment (jasonrclark) [inline public/index.html]: Do we know which part of the build machinery re-inserts this for us? Is it the npm buld, or gh-pages? @eknuth

**Comment by jasonrclark** [comment]
To get things moving, this is ready enough to merge. Will continue separating pieces out in smaller, follow-up PRs.

File: public/index.html (hunk 3 above)
Comment (eknuth) [reply public/index.html]: @jasonrclark that is done by webpack through the npm react-scripts build step. All the files are built and staged in the build directory. The gh-pages deploy script copies that directory up to github, but that is all.

Comment (jasonrclark) [reply public/index.html]: Awesome, thanks for clarifying for me @eknuth!

**Pull request merged**